src/mesido/workflows/io/write_output.py

```diff
@@ -1491,21 +1493,36 @@ def _name_to_asset(name):
         #         )
         #     esh.save(str(filename))
 
-    def _write_json_output(self):
-        # TODO: still add solver stats as json output
-        results = self.extract_results()
+    def _write_json_output(
+        self,
+        results: AliasDict | Dict,
+        parameters: AliasDict | Dict,
+        bounds: AliasDict | Dict,
+        aliases: OrderedDict,
+        solver_stats: Dict,
+    ):
+        """
+        The results, parameters, bounds are saved as json files which can be used for further
+        processing. Aliases are also saved as this allows us to only save the necessary variables.
+        :param results: dictionary or Alias dictionary with the results of the optimization problem
+        :param parameters: dictionary or Alias dictionary with the parameters of the optimization
+        problem
+        :param bounds: dictionary or Alias dictionary with the bounds of the optimization problem
+        :param aliases: Alias dictionary describing all the aliases for the variables used
+        :param solver_stats: solver statistics provided by the solver
+        :return:
+        """
+
         workdir = self.output_folder
 
-        parameters = self.parameters(0)
-        parameters_dict = dict()
+        parameters_dict = parameters
```

Comment:
In theory the loop below should not be needed, but in case that there end up other parameters then just floats it is important to have. Have changed it indeed back to parameters_dict = dict()